### PR TITLE
Added support for MKS Gen L v2.0 and v2.1 boards

### PR DESCRIPTION
**Pull request by CTetford**
-Make LCD pins configurable 
-Add UART support for MKS Gen L
-bug fixes

### CODE REVIEW

File: Software/Arduino code/OpenAstroTracker/src/b_setup.hpp [hunk 1]
```diff
@@ -8,10 +8,10 @@
 #include "Utility.hpp"
 #include "EPROMStore.hpp"
 #include "inc/Config.hpp"
-
+#include "../Configuration_pins.hpp"
```
Comment (andre-stefanov) [inline Software/Arduino code/OpenAstroTracker/src/b_setup.hpp]: please change to `inc/Config.hpp`

File: Software/Arduino code/OpenAstroTracker/src/LcdMenu.cpp [hunk 2]
```diff
@@ -1,14 +1,15 @@
 #include "Utility.hpp"
 #include "EPROMStore.hpp"
 #include "LcdMenu.hpp"
+#include "../Configuration_pins.hpp"
```
Comment (andre-stefanov) [inline Software/Arduino code/OpenAstroTracker/src/LcdMenu.cpp]: please change to `inc/Config.hpp`

**Review by andre-stefanov — CHANGES_REQUESTED** [review]

File: Software/Arduino code/OpenAstroTracker/Configuration_pins.hpp [hunk 3]
```diff
@@ -57,6 +57,18 @@
...
+    
+    //LCD pin definitions (to allow configuration on different boards)
+    #if DISPLAY_TYPE == DISPLAY_TYPE_LCD_KEYPAD
+        #define LCD_PINA0 0
+        #define LCD_PIN4 4
+        #define LCD_PIN5 5
+        #define LCD_PIN6 6
+        #define LCD_PIN7 7
+        #define LCD_PIN8 8
+        #define LCD_PIN9 9
```
Comment (andre-stefanov) [inline Software/Arduino code/OpenAstroTracker/Configuration_pins.hpp:69]: Please name these definitions by actual pin names on the LCD shield

**Review by andre-stefanov — CHANGES_REQUESTED** [review]

**Comment by andre-stefanov** [comment]
Don't merge this PR yet. i will maybe have to refactor the code a bit to fit in our coding conventions and/or best practices

File: Software/Arduino code/OpenAstroTracker/Configuration_pins.hpp (hunk 3 above)
Comment (CTetford) [reply Software/Arduino code/OpenAstroTracker/Configuration_pins.hpp:79]: The pins on the LCD shields are not labelled. The next best reference would be their intended pins on the Arduino Mega they are designed for.

File: Software/Arduino code/OpenAstroTracker/Configuration_adv.hpp [hunk 4]
```diff
@@ -152,6 +153,10 @@
 #define ALTITUDE_ARC_SECONDS_PER_STEP (0.61761f)
 #define ALTITUDE_STEPS_PER_ARC_MINUTE (60.0f/ALTITUDE_ARC_SECONDS_PER_STEP)
 
+// Enable dew heater output (for boards that have MOSFETs)
+#define DEW_HEATER 1
```
Comment (ClutchplateDude) [inline Software/Arduino code/OpenAstroTracker/Configuration_adv.hpp:157]: This should be off by default

**Review by ClutchplateDude — CHANGES_REQUESTED** [review]

File: Software/Arduino code/OpenAstroTracker/Configuration_adv.hpp (hunk 4 above)
Comment (CTetford) [reply Software/Arduino code/OpenAstroTracker/Configuration_adv.hpp]: Agreed - this was leftover from testing.

**Review by ClutchplateDude — APPROVED** [review]

**Review by andre-stefanov — CHANGES_REQUESTED** [review]
This PR is still not building for several configurations. We need to fix them before merging.

**Review by andre-stefanov — APPROVED** [review]

**Pull request merged**